packages/esm-ward-app/src/ward-workspace/admit-patient-form-workspace/admit-patient-form-workspace.workspace.tsx

```diff
@@ -0,0 +1,174 @@
+import { assignPatientToBed, createEncounter } from '../../ward.resource';
+import type { WardConfigObject } from '../../config-schema';
+import { useInpatientRequest } from '../../hooks/useInpatientRequest';
+
+interface AdmitPatientFormWorkspaceProps extends DefaultWorkspaceProps {
+  patient: Patient;
+}
+
+const AdmitPatientFormWorkspace: React.FC<AdmitPatientFormWorkspaceProps> = ({
+  patient,
+  closeWorkspace,
+  closeWorkspaceWithSavedChanges,
+  promptBeforeClosing,
+}) => {
+  const { t } = useTranslation();
+  const { location } = useWardLocation();
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const { mutate: mutateInpatientRequest } = useInpatientRequest();
+  const { admissionEncounterTypeUuid } = useConfig<WardConfigObject>();
+  const [showErrorNotifications, setShowErrorNotifications] = useState(false);
+  const { isLoading, admissionLocation, mutate: mutateAdmissionLocation } = useAdmissionLocation();
+  const beds = useMemo(() => (isLoading ? [] : filterBeds(admissionLocation)), [admissionLocation]);
+  const getBedRepresentation = useCallback((bedLayout: BedLayout) => {
+    const bedNumber = bedLayout.bedNumber;
+    const patients =
+      bedLayout.patients.length === 0
+        ? [t('emptyBed', 'Empty')]
+        : bedLayout.patients.map((patient) => patient?.person?.preferredName?.display);
+    return [bedNumber, ...patients].join(' · ');
+  }, []);
+
+  const zodSchema = useMemo(
+    () =>
+      z.object({
+        bedId: z.number({
+          required_error: t('bedSelectionRequired', 'Please select a bed to move forward'),
+        }),
+      }),
+    [],
+  );
+
+  type FormValues = z.infer<typeof zodSchema>;
+
+  const {
+    control,
+    formState: { errors, isDirty },
+    handleSubmit,
+    getValues,
+    watch,
+  } = useForm<FormValues>({
+    resolver: zodResolver(zodSchema),
+  });
+
+  useEffect(() => {
+    promptBeforeClosing(() => isDirty);
+  }, [isDirty]);
+
+  const onSubmit = useCallback(
+    (values: FormValues) => {
+      setShowErrorNotifications(false);
+      setIsSubmitting(true);
+      const bedSelected = beds.find((bed) => bed.bedId === values.bedId);
+      createEncounter(patient.uuid, admissionEncounterTypeUuid, location?.uuid)
+        .then(async (response) => {
+          if (response.ok) {
+            return assignPatientToBed(values.bedId, patient.uuid, response.data.uuid);
+          }
+        })
+        .then(() => {
+          setIsSubmitting(false);
+          showSnackbar({
+            kind: 'success',
+            title: t('patientAdmittedSuccessfully', 'Patient admitted successfully'),
+            subtitle: t(
+              'patientAdmittedSuccessfullySubtitle',
+              '{{patientName}} has been successfully admitted to the bed {{bedNumber}}',
+              {
+                patientName: patient.person.preferredName.display,
+                bedNumber: bedSelected.bedNumber,
+              },
+            ),
+          });
```

Comment:
Can we add some error messages alerting user what happened in either of the two failure scenarios I'm seeing:
1) The whole request failed
2) The admission encounter was created, but the bed assignment failed.

Reply:
What should be the error message for the second case @mogoodrich ?

Reply:
We discussed this and came up with something like "Patent was admitted but bed assignment failed".  We can discuss with team and tweak the message if necessary, but this seems fine to start.